src/main/java/com/currencyfair/onesignal/model/notification/Notification.java

```diff
@@ -193,7 +194,7 @@
      * <p>
      * Example: {@code {"abc": "123", "foo": "bar"}}
      */
-    private Map<String, String> data;
+    private JsonNode data;
```

Comment:
Any specific reason to switch from Map to JsonNode?

Reply:
There are two main reasons to do that. First of all, I just want to pass value different than String like Object, int, boolean etc. Secondy, I am trying to send complex json via data and it is hard to create it with using map and also according to documentation, data is object so it seems to me JsonNode is better option but at that moment, we break the backward compatibility. I am open to suggestion and I can update pull request as your recommendation.

Reply:
I agree that Map<String, String> significantly limits functionality. Usage of Map<String, Object> may minimize impact. But it is still not fully backwards compatible.

Reply:
Fair enough. Some of the code has changed, @karayel if you can pull the latest changes and up the code, I am happy to merge the changes.

Reply:
BTW, the `externalUserIds` is already available https://github.com/CurrencyFair/OneSignal-Java-SDK/blob/master/src/main/java/com/currencyfair/onesignal/model/notification/Notification.java#L101